### PR TITLE
Automatically advance the screen if exactly 1 ID scan match

### PR DESCRIPTION
**Pull request by kshen0**
## Overview

https://github.com/votingworks/vxsuite/issues/6633

When an ID is scanned and there is exactly 1 match for first name, middle name, last name, AND suffix, jump directly to the check-in page (for pollworkers) or voter information screen (for election managers).

## Demo Video or Screenshot

See Slack
https://votingworks.slack.com/archives/C07CR6T1TLK/p1750785008365259

## Testing Plan

* Manually tested with barcode scanner hardware
* Added e2e test
* Some additions are uncovered. I added `istanbul ignore next` to some lines new and existing to bring the cover up to existing thresholds. I'll need to go back and add coverage for some of the edge cases after the feature work is done.

## Checklist

- [x] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [x] I have added a screenshot and/or video to this PR to demo the change
- [x] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: apps/pollbook/frontend/src/poll_worker_screen.tsx [hunk 1]
```diff
@@ -116,12 +139,11 @@ export function VoterCheckInScreen(): JSX.Element | null {
     case 'search':
       return (
         <VoterSearchScreen
+          includeInactiveVoters={false}
```
Comment (carolinemodic) [inline apps/pollbook/frontend/src/poll_worker_screen.tsx:142]: remove look like this snuck in with the merge

File: apps/pollbook/frontend/src/voter_search_screen.tsx [hunk 2]
```diff
@@ -62,49 +59,94 @@ const VoterTable = styled(Table)`
...
   };
 }
 
 export function VoterSearch({
   search,
   setSearch,
+  // Function to call when exactly one voter is matched by scanning an ID
+  onBarcodeScanMatch,
   renderAction,
+  includeInactiveVoters,
```
Comment (carolinemodic) [inline apps/pollbook/frontend/src/voter_search_screen.tsx:80]: remove

File: apps/pollbook/frontend/src/voter_search_screen.tsx [hunk 3]
```diff
@@ -224,18 +267,28 @@ export function CheckInDetails({
 }
 
 export function VoterSearchScreen({
+  includeInactiveVoters,
```
Comment (carolinemodic) [inline apps/pollbook/frontend/src/voter_search_screen.tsx:267]: remove

**Review by carolinemodic — APPROVED** [review]
overall lgtm but there are some includeInactiveVoters remnants that crept back in
